remote/src/main/scala/org/apache/pekko/remote/transport/netty/TcpSupport.scala

```diff
@@ -56,8 +58,16 @@ private[remote] trait TcpHandlers extends CommonHandlers {
   }
 
   override def onMessage(ctx: ChannelHandlerContext, msg: ByteBuf): Unit = {
-    val bytes: Array[Byte] = ByteBufUtil.getBytes(msg)
-    if (bytes.length > 0) notifyListener(ctx.channel(), InboundPayload(ByteString(bytes)))
+    try {
+      val bytes: Array[Byte] = ByteBufUtil.getBytes(msg)
+      if (bytes.length > 0) notifyListener(ctx.channel(), InboundPayload(ByteString(bytes)))
+    } catch {
+      case NonFatal(e) =>
+        log.warning("Error while handling message from [{}]: {}", ctx.channel().remoteAddress(), e)
+        onException(ctx, e)
+    } finally {
+      msg.release()
```

Comment:
Sorry for not checking the `LengthFieldBasedFrameDecoder` in detail, we need to turn on the leak detector in nightly testing I think.

Reply:
I can't call release, because it's a `retainedSlice` not a `readRetainedSlice`, will dig it deeper